### PR TITLE
add back grad_norm metric

### PR DESCRIPTION
**Pull request by kashif**
add back the grad_norm and skipped_step metric via the new log api in torchtitan

## Summary by CodeRabbit

This release delivers improvements to system stability and training process monitoring:

- **Chores**
  - Upgraded a core third-party dependency to the latest version to bolster overall performance and reliability.

- **New Features**
  - Enhanced training logging to capture additional diagnostic metrics, including gradient norms, last learning rate, and estimated time of arrival, providing more comprehensive insights during training sessions.

### CODE REVIEW

File: flame/train.py [hunk 1]
```diff
@@ -729,7 +729,12 @@ def main(job_config: JobConfig):
                 train_state.global_max_losses.append(global_max_loss)
 
                 # Log using the metric processor
-                metric_logger.log(train_state.step, global_avg_loss, global_max_loss)
+                metric_logger.log(
+                    train_state.step,
+                    global_avg_loss,
+                    global_max_loss,
+                    extra_metrics={"loss_metrics/grad_norm": grad_norm.item()},
```
Comment (yzhangcs) [inline flame/train.py:736]: How about adding some rounding for grad_norm, e.g., .4f?

**Review by yzhangcs — APPROVED** [review]

File: flame/train.py (hunk 1 above)
Comment (kashif) [reply flame/train.py:736]: ah sure!

Comment (kashif) [reply flame/train.py]: actually, this is not used for printing but logging via e.g. wandb and there the formatting is not an issue

File: flame/train.py [hunk 2]
```diff
@@ -729,7 +729,15 @@ def main(job_config: JobConfig):
...
 
                 # Log using the metric processor
-                metric_logger.log(train_state.step, global_avg_loss, global_max_loss)
+                metric_logger.log(
+                    train_state.step,
+                    global_avg_loss,
+                    global_max_loss,
+                    extra_metrics={
+                        "optimizer/grad_norm": grad_norm.item(),
+                        "optimizer/skipped_step": train_state.skipped_step,
```
Comment (yzhangcs) [inline flame/train.py:738]: @kashif Could you also add sth like "elapsed time<eta" in some format, which I think would be  good for some impatient guys like me.

Comment (kashif) [reply flame/train.py:738]: ok let me see

**Comment by yzhangcs** [comment]
local running results:
image

**Comment by yzhangcs** [comment]
The logging looks good enough to me for now. 
Maybe we need one line log display per step, which could be achieved by making PRs to torchtitan
